tests/test_utils.py

```diff
@@ -29,15 +29,23 @@ def test_default_device(self):
         self.assertEqual(default_device(user).pk, default.pk)
 
     def test_backup_phones(self):
-        self.assertQuerysetEqual(list(backup_phones(None)),
-                                 list(PhoneDevice.objects.none()))
         user = self.create_user()
         user.phonedevice_set.create(name='default', number='+12024561111')
         backup = user.phonedevice_set.create(name='backup', number='+12024561111')
-        phones = backup_phones(user)
 
-        self.assertEqual(len(phones), 1)
-        self.assertEqual(phones[0].pk, backup.pk)
+        with self.settings(
+            TWO_FACTOR_CALL_GATEWAY='two_factor.gateways.fake.Fake',
+        ):
+            self.assertQuerysetEqual(list(backup_phones(None)),
+                                    list(PhoneDevice.objects.none()))
+            phones_with_gw = backup_phones(user)
+
+            self.assertEqual(len(phones_with_gw), 1)
+            self.assertEqual(phones_with_gw[0].pk, backup.pk)
+
+        phones_without_gw = backup_phones(user)
+
+        self.assertEqual(len(phones_without_gw), 0)
```

Comment:
This took me way too long to work out what was actually being tested here. These should probably be three separate tests for each of the conditions we're testing here, but I'd accept a short comment above each `backup_phones`

Reply:
i rewrote it as a parameterised test (it's in the "new" place `tests.test_utils.PhoneUtilsTests.test_backup_phones`)